### DISCUSSION TITLE
Process for firmware upgrade for systemd uefi boot without shim.

### DISCUSSION THREAD

**Discussion by sagarchalise** [fwupd/fwupd#2772] [Q&A]
Will disabling shim work for upgrade or do I need to do something extra if I am booting from [systemd-boot](https://wiki.archlinux.org/index.php/systemd-boot) ?

**Comment by superm1** [comment]
Will you be using secure boot?

**Reply by sagarchalise** [reply]
yes, I am using secure boot with preloader and hashtool to sign the loader.

**Reply by superm1** [reply]
Without shim you will need to directly sign the fwupd binary.

**Reply by superm1** [reply]
Which FYI I think is not how preloader works.  If you want preloader to work, I think you will need to do some code changes to fwupd so that the boot entries are created properly.  PR definitely welcome in this area if you want to work on it.
